gapis/resolve/memory.go

```diff
@@ -309,3 +320,73 @@ func MemoryAsType(ctx context.Context, p *path.MemoryAsType, rc *path.ResolveCon
 
 	return vals[0], nil
 }
+
+// Function memoryAsType resolves typed memory related raw data,
+// and return the decoded memory.
+// This is a private resolving function inside gapis,.
+func memoryAsType(ctx context.Context, s *api.GlobalState, rng memory.Range, poolId uint32, typeIndex uint64,
+	p path.Node, rc *path.ResolveConfig) (*memory_box.Value, error) {
+	// Check whether the requested pool was ever created.
+	pool, err := s.Memory.Get(memory.PoolID(poolId))
```

Comment:
Should we pass the pool from the outside rather than the id? We have it.

Reply:
done in the new commit.